File: backend/routes/configuracion.js
const express = require('express');
const router = express.Router();
const supabase = require('../config/database');

// Obtener configuración
router.get('/', async (req, res) => {
  try {
    const { data, error } = await supabase
      .from('configuracion')
      .select('*')
      .single();

    if (error && error.code !== 'PGRST116') throw error;
    
    // Si no existe configuración, crear una por defecto
    if (!data) {
      const { data: nuevaConfig, error: createError } = await supabase
        .from('configuracion')
        .insert([{
          nombre_empleado: 'Empleado',
          tema_oscuro: false,
          garantia_default: 50.0,
          mora_diaria: 10.0,
          dias_maximos_mora: 7
        }])
        .select()
        .single();

      if (createError) throw createError;
      return res.json(nuevaConfig);
    }

    res.json(data);
  } catch (error) {
    res.status(500).json({ error: error.message });
  }
});

// Actualizar configuración
router.put('/', async (req, res) => {
  try {
    const { 
      nombre_empleado, 
      tema_oscuro, 
      garantia_default, 
      mora_diaria, 
      dias_maximos_mora 
    } = req.body;

    // Verificar si existe configuración
    const { data: existente } = await supabase
      .from('configuracion')
      .select('id')
      .single();

    let data, error;

    if (existente) {
      // Actualizar
      const result = await supabase
        .from('configuracion')
        .update({
          nombre_empleado,
          tema_oscuro,
          garantia_default,
          mora_diaria,
          dias_maximos_mora
        })
        .eq('id', existente.id)
        .select()
        .single();

      data = result.data;
      error = result.error;
    } else {
      // Crear
      const result = await supabase
        .from('configuracion')
        .insert([{
          nombre_empleado,
          tema_oscuro,
          garantia_default,
          mora_diaria,
          dias_maximos_mora
        }])
        .select()
        .single();

      data = result.data;
      error = result.error;
    }

    if (error) throw error;
    res.json(data);
  } catch (error) {
    res.status(500).json({ error: error.message });
  }
});

module.exports = router;

File: flutter_app/lib/models/configuracion.dart
class Configuracion {
  final String? id;
  final String nombreEmpleado;
  final bool temaOscuro;
  final double garantiaDefault;
  final double moraDiaria;
  final int diasMaximosMora;

  Configuracion({
    this.id,
    required this.nombreEmpleado,
    required this.temaOscuro,
    required this.garantiaDefault,
    required this.moraDiaria,
    required this.diasMaximosMora,
  });

  factory Configuracion.fromJson(Map<String, dynamic> json) {
    return Configuracion(
      id: json['id'],
      nombreEmpleado: json['nombre_empleado'] ?? 'Empleado',
      temaOscuro: json['tema_oscuro'] ?? false,
      garantiaDefault: double.parse(
        json['garantia_default']?.toString() ?? '50.0',
      ),
      moraDiaria: double.parse(json['mora_diaria']?.toString() ?? '10.0'),
      diasMaximosMora: json['dias_maximos_mora'] ?? 7,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'nombre_empleado': nombreEmpleado,
      'tema_oscuro': temaOscuro,
      'garantia_default': garantiaDefault,
      'mora_diaria': moraDiaria,
      'dias_maximos_mora': diasMaximosMora,
    };
  }

  Configuracion copyWith({
    String? nombreEmpleado,
    bool? temaOscuro,
    double? garantiaDefault,
    double? moraDiaria,
    int? diasMaximosMora,
  }) {
    return Configuracion(
      id: id,
      nombreEmpleado: nombreEmpleado ?? this.nombreEmpleado,
      temaOscuro: temaOscuro ?? this.temaOscuro,
      garantiaDefault: garantiaDefault ?? this.garantiaDefault,
      moraDiaria: moraDiaria ?? this.moraDiaria,
      diasMaximosMora: diasMaximosMora ?? this.diasMaximosMora,
    );
  }
}
